mlrun/api/crud/projects.py

```diff
@@ -286,22 +286,33 @@ async def _get_project_resources_counters(
             self._cache["project_resources_counters"]["ttl"] = ttl_time
         return self._cache["project_resources_counters"]["result"]
 
+    @staticmethod
+    def _list_pipelines(
+        session,
+        format_: mlrun.api.schemas.PipelinesFormat = mlrun.api.schemas.PipelinesFormat.metadata_only,
+    ):
+        return mlrun.api.crud.Pipelines().list_pipelines(session, "*", format_=format_)
+
     async def _calculate_pipelines_counters(
         self,
-    ) -> typing.Dict[str, int]:
-        def _list_pipelines(session):
-            return mlrun.api.crud.Pipelines().list_pipelines(
-                session, "*", format_=mlrun.api.schemas.PipelinesFormat.metadata_only
-            )
-
-        project_to_running_pipelines_count = collections.defaultdict(int)
+    ) -> typing.Dict[str, typing.Union[int, None]]:
+        project_to_running_pipelines_count = collections.defaultdict(lambda: 0)
```

Comment:
add a comment explaining why you used a `collections.defaultdict` and what it is